internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -0,0 +1,1579 @@
+		internaltypes.IsDefined(model.CibaRequestObjectSigningAlgorithm) ||
+		internaltypes.IsDefined(model.CibaUserCodeSupported)) {
+		resp.Diagnostics.AddError("ciba attributes can only be configured when \"CIBA\" is included in grant_types.", "")
+	}
+	if hasCibaGrantType &&
+		(internaltypes.IsDefined(model.CibaDeliveryMode) &&
+			model.CibaDeliveryMode.ValueString() == "PING" &&
+			!internaltypes.IsDefined(model.CibaNotificationEndpoint)) {
+		resp.Diagnostics.AddError("ciba_notification_endpoint must be defined when ciba_delivery_mode is \"PING\".", "")
+	}
+
+	// Client Auth Validation
+	// ID Token Signing Algorithm Validation when client_auth is not defined
+	if !internaltypes.IsDefined(model.ClientAuth) {
+		var algorithmAttributeSet []string
+		if internaltypes.IsDefined(model.OidcPolicy) && model.OidcPolicy.Attributes()["id_token_signing_algorithm"] != nil {
+			algorithmAttributeSet = append(algorithmAttributeSet, model.OidcPolicy.Attributes()["id_token_signing_algorithm"].(types.String).ValueString())
+		}
+
+		if internaltypes.IsDefined(model.TokenIntrospectionSigningAlgorithm) {
+			algorithmAttributeSet = append(algorithmAttributeSet, model.TokenIntrospectionSigningAlgorithm.ValueString())
+		}
+
+		if internaltypes.IsDefined(model.JwtSecuredAuthorizationResponseModeSigningAlgorithm) {
+			algorithmAttributeSet = append(algorithmAttributeSet, model.JwtSecuredAuthorizationResponseModeSigningAlgorithm.ValueString())
+		}
+
+		for _, algorithmVal := range algorithmAttributeSet {
+			if algorithmVal == "HS256" {
+				resp.Diagnostics.AddError("client_auth must be defined when using the \"HS256\" signing algorithm", "")
+			}
+		}
+
+		if internaltypes.IsDefined(model.TokenIntrospectionEncryptionAlgorithm) {
+			resp.Diagnostics.AddError("client_auth must be configured when token_introspection_encryption_algorithm is configured.", "")
+		}
+	}
+
+	// Restrict Scopes Validation
+	if len(model.RestrictedScopes.Elements()) > 0 && !model.RestrictScopes.ValueBool() {
+		resp.Diagnostics.AddError("restrict_scopes must be set to true to configure restricted_scopes.", "")
+	}
+
+	// OIDC Policy Validation
+	if internaltypes.IsDefined(model.OidcPolicy) {
+		oidcPolicy := model.OidcPolicy.Attributes()
+		pairwiseIdentifierUserType := oidcPolicy["pairwise_identifier_user_type"]
+		oidcPolicySectorIdentifierUri := oidcPolicy["sector_identifier_uri"]
+		if (pairwiseIdentifierUserType != nil && !pairwiseIdentifierUserType.(types.Bool).ValueBool()) && internaltypes.IsDefined(oidcPolicySectorIdentifierUri) {
+			resp.Diagnostics.AddError("sector_identifier_uri can only be configured when pairwise_identifier_user_type is set to true.", "")
+		}
+	}
+
+	// JWKS Settings Validation
+	if !internaltypes.IsDefined(model.JwksSettings) {
+		if internaltypes.IsDefined(model.TokenIntrospectionEncryptionAlgorithm) {
+			resp.Diagnostics.AddError("token_introspection_encryption_algorithm must not be configured jwks_settings is configured.", "")
```

Comment:
This message seems wrong - if we get to here, then we've confirmed that jwks_settings is *not* configured

Reply:
I got the message based on copilot, I will reword it

Reply:
re-worded